### ISSUE TITLE
Your claim that Neural Cleanse needs 'extra requirements' is inexact and misleading.

### ISSUE THREAD

**Issue by LeeBC2298** [SCLBD/DBD#2]
I think your comparison with Neural Cleanse is unfair.
我认为您和NC的对比并不公平。

Neural Cleanse only uses a clean dataset. However, your method needs the poisoned training dataset.
NC只需要一个干净的小数据集。然而您的方法需要完整的投毒训练集。

In fact these are totally two different scenarios, it is hard to say which setting is simple or needs extra requirements.
我认为这完全是两个不同的场景，很难说哪个场景下需要的额外信息更多。

If you have to compare, I think your method requires more 'defender capability' in practice.
如果硬要比的话，我也认为在实际应用中，您的方法比NC实现起来困难。

Your claim that Neural Cleanse needs 'extra requirements' is inexact and misleading.
您说NC需要额外的信息，这并不确切，而且具有误导性。

**Comment by SCLBD** [comment]
Thanks for this feedback. Let's compare them as follows:

The input of DBD: a unverified third-party training set (may include poisoned samples, may not)
The input of NC: a backdoored model, a clean subset of the original training set

The output of DBD: a secure model, a clean training set
The output of NC: a secure model

The goals of DBD and NC are same, with different inputs. But I don't think they are "totally two different scenarios". 
DBD's setting is very practical, when you download a third party dataset, you cannot ensure whether there are poisons or not. DBD provides a secure training method. 

For the clean subset, it is not only required in NC, but also in FP, NAD, ANP. It is a common setting in some defense methods, but doesn't mean it is practical. We don't think it is inappropriate to call it "extra requirement". 

We have provided a clear comparisons of different settings between existing defense methods in another work (see Table 2 in https://arxiv.org/pdf/2206.12654.pdf) 

We keep open to further discuss with you.

**Comment by LeeBC2298** [comment]
Firstly, the problem is that you don't admit that these are two completely different settings, but bring Neural Cleanse into your settings for comparison.
首先，问题就在于您不肯承认这是两个不同的设定，而一定要将NC带入到您的场景下去使用。

What if in a MLaaS scenery, users could only get the pretrained model? Then the user or some third-party defender could not get the original training set, they could only get their own eval dataset or some clean data from the Internet.
如果在MLaaS的场景下，用户只能获取预训练的模型呢？此时用户或者专门提供模型防御的第三方无法获取原训练数据集，只能利用自己的评测数据集或互联网上的干净数据。

Secondly, Neural Cleanse doesn't need 'a clean subset of the original training data'. It only needs some clean data for the same task. For example, when checking wheather an ImageNet trained model is poisoned or not, Neural Cleanse could use any dog or cat images from any sources (of course, the premise is to ensure clean), not only dog or cat images from the ImageNet Dataset. 
其次，NC不需要“原训练数据集的干净子集”，它只需要同任务的干净数据。举个例子，如果检测一个ImageNet训练的模型是否遭受投毒攻击，NC只需要用任何来源的猫或狗的图像（当然，前提是保证干净）就可以了，并不一定要用ImageNet数据集中的猫狗图像。

Not needing the original training dataset is a great advantage of Neural Cleanse, and I think your claim that it requires 'a clean subset of the original training dataset' is inexact and misleading.
脱离训练数据集是NC的一大优点，我认为您说它“需要训练数据集的干净子集”既不确切，而且有误导性。

**Comment by SCLBD** [comment]
我们很欢迎客观的学术讨论，而不是批判式的扣帽子，如果你觉得对额外数据集的评价不合理，可以在你的学术论文中写出来，供评审和观众评价。

我们不觉得把NC作为对比方法有任何不妥，不觉得额外信息的评价会损坏或提升NC的性能，其他研究者也有自己的判断，不至于被误导对NC有什么不合理的评价。

把时间放在研究更好的方法上吧。

**Comment by SCLBD** [comment]
留给读者评论吧

**Comment by LeeBC2298** [comment]
I don't think your method is inferior to NC. After all, the experimental results show that your method does achieve better network repair effect than NC under the premise that the accuracy of the clean dataset is comparable. And your experimental scenario is one of many reasonable settings for backdoor attacks.
我并不觉得您的方法不如NC，毕竟实验结果上说明，在干净数据集精度相当的前提下，您的方法的确比NC做到更好的网络修复效果。并且您的实验场景也是后门攻击的众多合理设定下的一种。

I think it would be nice to objectively claim that NC is in its own setting, but I just don't agree with the emphasis that NC "uses additional data". After all, NC itself was not designed for your setting.
我认为客观地将NC列为其他设定下的方法同您的方法对比就很好了，但是在行文上强调该方法“利用额外数据”，我实在无法认同。毕竟NC本身并不是为了该设定而设计的。

In addition, I hope to get an answer to the last two questions I raised.
另外，我最后提出的两点问题，还希望得到正面答复。

**Comment by SCLBD** [comment]
我们论文中所写的是我们认为正确的学术评价，是否准确是我们的学术评判能力问题，观众也会有判断。如果你觉得不正确，我建议可以把你的观点写到将来的学术论文中。

**Comment by LeeBC2298** [comment]
这只是针对这篇文章观点的讨论而已，我并不觉得对某篇论文中 对另某篇论文的评价 的评价适合写进论文里。就当您是在为难我了。
